### PR TITLE
Define CRDs for HTTPAPI and Quota spec and distribute to proxies

### PR DESCRIPTION
**Pull request by ayj**
1) create corresponding CRD for HTTPAPISpec, HTTPAPISpecBinding, QuotaSpec, and QuotaSpecBinding (see https://github.com/istio/api/blob/master/mixer/v1/config/client/api_spec.proto)

2) add corresponding validation code for new model types

3) update `istioConfigStore` to select HTTPAPISpec and QuotaSpec config associated with destination service instances using the corresponding HTTPAPISpec and QuotaSpec bindings.

4) distribute the HTTPAPI and Quota spec to proxy as part of mixerclient filter configuration.

### CODE REVIEW

**Comment by ayj** [comment]
That that this is a continuation of https://github.com/istio/old_pilot_repo/pull/1644 which was rebased from istio/pilot onto istio/istio.

File: pilot/model/config.go [hunk 1]
```diff
@@ -530,3 +576,117 @@ func RejectConflictingEgressRules(egressRules map[string]*proxyconfig.EgressRule
 
 	return filteredEgressRules, errs
 }
+
+// `istio.mixer.v1.config.client.IstioService` and
+// `istio.proxy.v1.config.IstioService` are logically
+// equivalent. Convert from mixer-to-proxy representation so we can
+// use ResolveHostname below.
```
Comment (kyessenov) [inline pilot/model/config.go]: note: `IstioService` is going away in pilot (see @louiscryan and @rshriram latest gateway proposal)

File: pilot/model/conversion.go [hunk 2]
```diff
@@ -31,7 +32,10 @@ import (
 func (ps *ProtoSchema) Make() (proto.Message, error) {
 	pbt := proto.MessageType(ps.MessageName)
 	if pbt == nil {
-		return nil, fmt.Errorf("unknown type %q", ps.MessageName)
+		pbt = gogoproto.MessageType(ps.MessageName)
```
Comment (kyessenov) [inline pilot/model/conversion.go]: clarify where gogoproto comes in from?

File: pilot/model/validation.go [hunk 3]
```diff
@@ -1222,3 +1226,210 @@ func ValidateProxyConfig(config *proxyconfig.ProxyConfig) (errs error) {
...
+			if m.Cookie == "" {
+				errs = multierror.Append(errs, errors.New("cookie cannot be empty"))
+			}
+		}
+	}
+	return errs
+}
+
+// ValidateHTTPAPISpecBinding checks that HTTPAPISpecBinding is well-formed.
+func ValidateHTTPAPISpecBinding(msg proto.Message) error {
```
Comment (kyessenov) [inline pilot/model/validation.go]: note: we're entertaining the idea of using protoc-validate annotations from lyft to automate these functions. this is fine for now.

**Comment by ayj** [comment]
Agreed 100% on consistent codegen for our api-machinery. cc @gjwestlife

**Pull request merged**